zerver/lib/actions.py

```diff
@@ -3465,6 +3465,28 @@ def send_peer_subscriber_events(
         altered_user_ids = altered_user_dict[stream_id]
         peer_user_ids = private_peer_dict[stream_id] - altered_user_ids
 
+        # For sending user left stream messages
+        if op == "peer_remove":
+            notifications = []
+            for user_id in altered_user_ids:
+                try:
+                    user_profile = get_user_by_id_in_realm_including_cross_realm(user_id, realm)
+                except UserProfile.DoesNotExist:  # nocoverage
+                    raise JsonableError(_("Invalid user ID {}").format(user_id))  # nocoverage
+
+                if user_profile.is_bot:
+                    continue  # nocoverage
+
+                msg = "{} left {}.".format(user_profile.full_name, stream_dict[stream_id].name)
```

Comment:
We don't usually use plain text for strings for reasons of localization. Please take a look at [i18n](https://zulip.readthedocs.io/en/latest/translating/internationalization.html?highlight=internalization) to support translation. I think we might want something like `_("{...} left {..}").format(...)` here.